ui/src/app/components/copy-modal.tsx

```diff
@@ -156,40 +161,37 @@ class CopyNotebookModal extends React.Component<Props, State> {
       <div>
         <div style={headerStyles.formLabel}>Destination *</div>
         <Select
-            value={this.state.destination}
-            options={this.state.writeableWorkspaces.map(workspace => ({
-              'value': workspace,
-              'label': workspace.name
-            }))}
-            onChange={(value) => { this.setState({ destination: value }); }} />
+          value={this.state.destination}
+          options={this.state.writeableWorkspaces.map(workspace => ({
+            'value': workspace,
+            'label': workspace.name
+          }))}
+          onChange={(value) => { this.setState({ destination: value }); }} />
         <div style={headerStyles.formLabel}>Name *</div>
         <TextInput
-            autoFocus
-            value={this.state.newName}
-            onChange={v => this.setState({ newName: v })}
+          autoFocus
+          value={this.state.newName}
+          onChange={v => this.setState({ newName: v })}
         />
         {this.state.requestState === RequestState.ERROR &&
-          <ValidationError> {this.state.errorMsg} </ValidationError>}
+        <ValidationError> {this.state.errorMsg} </ValidationError>}
       </div>
     );
   }
 
-  renderErrorBody() {
-    return <div style={headerStyles.formLabel}>{this.state.errorMsg}</div>;
-  }
-
   renderSuccessBody() {
+    const {fromResourceName, resourceType} = this.props;
     return (
       <div> Successfully copied
-      <b style={boldStyle}> {this.props.fromNotebook.name} </b> to
-      <b style={boldStyle}> {this.state.destination.name} </b>.
-      Do you want to view the copied notebook?</div>
+        <b style={boldStyle}> {fromResourceName} </b> to
+        <b style={boldStyle}> {this.state.destination.name} </b>.
+        Do you want to view the copied {ResourceTypeDisplayNames.get(resourceType)}?</div>
     );
   }
 }
 
 export {
-  CopyNotebookModal,
-  Props as CopyNotebookModalProps,
-  State as CopyNotebookModalState,
+  CopyModal,
+  Props as CopyModalProps,
+  State as CopyModalState,
```

Comment:
JW, is it possible to remove the state export? I think I added it before we decided thats probably bad practice.

Reply:
Probably. I will check.

Reply:
Nope, we need it for the tests.